k8s/Makefile

```diff
@@ -35,24 +35,35 @@ process-templates-inline: dist-path
 
 process-full-templates: process-templates
 	$(ENVSUBST_RUN) < $(K8S_TMPL_DIR)/service.yaml.tmpl > $(K8S_DIST_DIR)/service.yaml ; \
-	$(ENVSUBST_RUN) < $(K8S_TMPL_DIR)/ingress.yaml.tmpl > $(K8S_DIST_DIR)/ingress.yaml ;
+	$(ENVSUBST_RUN) < $(K8S_TMPL_DIR)/ingress.yaml.tmpl > $(K8S_DIST_DIR)/ingress.yaml
 
-deploy-k8s: validate-image process-templates
-	$(KUBECTL_SEARCH) apply --record -f $(K8S_DIST_DIR)/deployment.yaml
+deploy-k8s: check-k8s-cluster-var validate-image process-templates
+	$(KUBECTL_CMD) apply --record -f $(K8S_DIST_DIR)/deployment.yaml
 
-deploy-full-k8s: process-full-templates deploy-k8s
-	$(KUBECTL_SEARCH) apply --record -f $(K8S_DIST_DIR)/service.yaml ; \
-	$(KUBECTL_SEARCH) apply --record -f $(K8S_DIST_DIR)/ingress.yaml ; \
+deploy-full-k8s: check-k8s-cluster-var process-full-templates deploy-k8s
+	$(KUBECTL_CMD) apply --record -f $(K8S_DIST_DIR)/service.yaml ; \
+	$(KUBECTL_CMD) apply --record -f $(K8S_DIST_DIR)/ingress.yaml
 
-destroy-k8s:
-	$(KUBECTL_SEARCH) delete deploy ${DEPLOY_NAME} ;
+check-deploy: check-k8s-cluster-var process-templates
+	$(KUBECTL_CMD) rollout status -f $(K8S_DIST_DIR)/deployment.yaml
 
-destroy-full-k8s: destroy-k8s
-	$(KUBECTL_SEARCH) delete service ${DEPLOY_NAME}-service ; \
-	$(KUBECTL_SEARCH) delete ingress ${DEPLOY_NAME}-ingress ;
+check-deploy-with-rollback: check-k8s-cluster-var process-templates
+	$(KUBECTL_CMD) rollout status -f $(K8S_DIST_DIR)/deployment.yaml || { \
+	$(KUBECTL_CMD) get ev | grep $(DEPLOY_NAME) ; \
+	$(KUBECTL_CMD) rollout undo -f $(K8S_DIST_DIR)/deployment.yaml; exit 1; }
 
-run-k8s:
-	$(KUBECTL_SEARCH) run $(DEPLOY_NAME) --image=$(IMAGE_NAME) -i --rm --restart=Never $(K8S_RUN_ARGS) $(K8S_OVERRIDES)
+destroy-k8s: check-k8s-cluster-var
+	$(KUBECTL_CMD) delete deploy ${DEPLOY_NAME}
 
-logs-k8s:
-	$(KUBECTL_SEARCH) logs $(DEPLOY_NAME) -f
+destroy-full-k8s: check-k8s-cluster-var destroy-k8s
+	$(KUBECTL_CMD) delete service ${DEPLOY_NAME}-service ; \
+	$(KUBECTL_CMD) delete ingress ${DEPLOY_NAME}-ingress
+
+run-k8s: check-k8s-cluster-var
+	$(KUBECTL_CMD) run $(DEPLOY_NAME) --image=$(IMAGE_NAME) -i --rm --restart=Never $(K8S_RUN_ARGS) $(K8S_OVERRIDES)
+
+logs-k8s: check-k8s-cluster-var
+	$(KUBECTL_CMD) logs $(DEPLOY_NAME) -f
+
+check-k8s-cluster-var:
+	$(if $(value K8S_CLUSTER),,$(error "K8S_CLUSTER is required for k8s/Makefile"))
```

Comment:
I check variables with this: `@test ${ENV} || { echo ENV not set; }` and I can avoid "functions" only to do this, but this is just a FYI.

Reply:
🏆 

https://github.com/grupozap/authnetes/blob/master/Makefile#L47-L50